### PR TITLE
Chore/add button

### PR DESCRIPTION
**Pull request by salunkemanas**
Screenshot 2024-10-02 at 7 57 50 PM

### CODE REVIEW

File: src/styles/components/ui/Stack/Stack.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { ReactNode } from "react";
+
+type StackProps = {
+  children: ReactNode;
+  className?: string;
+  direction?: "row" | "column";
+  gap?: string;
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx]: add specific types `gap-xxs`, `gap-xs` ...`gap-4xl`

File: src/styles/components/ui/Stack/Stack.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ReactNode } from "react";
+
+type StackProps = {
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx:66]: Use PropsWithChildren

File: src/styles/components/ui/Stack/Stack.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { ReactNode } from "react";
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx]: import type

File: src/styles/components/ui/Stack/Stack.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  children: ReactNode;
+  className?: string;
+  direction?: "row" | "column";
+  gap?: string;
+};
+
+function Stack(props: StackProps) {
+  return (
+    <div
+      className={`flex flex-${props.direction} ${props.className} ${props.gap}`}
```
Comment (hatim-s) [inline src/styles/components/ui/Stack/Stack.tsx]: Use cva

File: src/styles/components/ui/Button/index.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+import { Button } from "../../base/button";
```
Comment (hatim-s) [inline src/styles/components/ui/Button/index.ts]: revert and make differernt pr

**Review by hatim-s — CHANGES_REQUESTED** [review]

File: src/styles/components/ui/Stack/Stack.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      "4xl": "gap-9",
+    },
+    gapY: {
+      default: "gap-y-0",
+      xxs: "gap-y-1",
+      xs: "gap-y-2",
+      sm: "gap-y-3",
+      md: "gap-y-4",
+      lg: "gap-y-6",
+      xl: "gap-6",
```
Comment (anandabhi04) [inline src/styles/components/ui/Stack/Stack.tsx]: gap-y.. and xl and lg hve same value

File: src/styles/components/ui/Stack/Stack.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      "4xl": "gap-9",
+    },
+    gapX: {
+      default: "gap-x-0",
+      xxs: "gap-x-1",
+      xs: "gap-x-2",
+      sm: "gap-x-3",
+      md: "gap-x-4",
+      lg: "gap-x-6",
+      xl: "gap-6",
```
Comment (anandabhi04) [inline src/styles/components/ui/Stack/Stack.tsx]: same

**Review by anandabhi04 — CHANGES_REQUESTED** [review]

**Review by anandabhi04 — APPROVED** [review]

**Review by hatim-s — APPROVED** [review]

**Pull request merged**